server/src/server.ts

```diff
@@ -115,6 +115,14 @@ let deleteProjectDiagnostics = (projectRootPath: string) => {
     projectsFiles.delete(projectRootPath);
   }
 };
+let sendCompilationFinishedMessage = () => {
+  let notification: m.NotificationMessage = {
+    jsonrpc: c.jsonrpcVersion,
+    method: "rescript/compilationFinished",
+  };
+  
+  send(notification);
```

Comment:
Is there a special reason for sending a custom notification rather than just calling ` customCommands.deadCodeAnalysisWithReanalyze` here?

Reply:
Yes, the reason for the notification itself is because my idea is to keep this feature contained inside of the extension code rather than the server. That has a few reasons:
* I didn't want to introduce state into the server (keeping track of whether the dead code analysis mode is activated or not)
* The code that controls the status bar item lives inside of the extension code itself, and not the server. I'm unsure whether this could live inside of the server at all and communicate via the LS protocol, but even if it could, I'm not sure whether I personally think it should live in the server or not. It's a pretty VSCode specific thing, at least at this point.
* It felt cleaner to separate the dead code diagnostics from the regular ReScript compiler diagnostics. That means it's easy to clear them independently of other diagnostics (which is done for example when stopping the dead code analysis mode, even if not all errors has been fixed).
* This way, any client using the language server can react to compilation finishing.

I'm very much open to trying to move all/most of the functionality into the server itself though, if we think that's a better place for it to live.

Reply:
Makes perfect sense, thanks for the explanation.